src/Analyzers/CSharp/Tests/UseThrowExpression/UseThrowExpressionTests.cs

```diff
@@ -570,5 +570,117 @@ public async Task TopLevelStatement()
 x = s ?? throw new ArgumentNullException();
 ", TestOptions.Regular.WithLanguageVersion(LanguageVersion.CSharp9));
         }
+
+        [WorkItem(38102, "https://github.com/dotnet/roslyn/issues/38102")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseThrowExpression)]
+        public async Task TestKeepCommnents1()
+        {
+            await TestAsync(
+@"using System;
+
+class Program
+{
+    static void Main() { }
+
+    private object _arg;
+
+    public Program(object arg)
+    {
+        if (arg == null) // Oh no
+        { 
+            [|throw|] new ArgumentNullException(nameof(arg));
+        }
+        _arg = arg;
+    }
+}",
+@"using System;
+
+class Program
+{
+    static void Main() { }
+
+    private object _arg;
+
+    public Program(object arg)
+    {
+        _arg = arg ?? throw new ArgumentNullException(nameof(arg));// Oh no
+    }
+}", TestOptions.Regular.WithLanguageVersion(LanguageVersion.CSharp9));
+        }
+
+        [WorkItem(38102, "https://github.com/dotnet/roslyn/issues/38102")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseThrowExpression)]
+        public async Task TestKeepCommnents2()
+        {
+            await TestAsync(
+@"using System;
+
+class Program
+{
+    static void Main() { }
+
+    private object _arg;
+
+    public Program(object arg)
+    {
+        if (arg == null)
+        {
+            // Oh no
+            [|throw|] new ArgumentNullException(nameof(arg));
+        }
+        _arg = arg;
+    }
+}",
+@"using System;
+
+class Program
+{
+    static void Main() { }
+
+    private object _arg;
+
+    public Program(object arg)
+    {
+        _arg = arg ?? throw new ArgumentNullException(nameof(arg));// Oh no
+    }
+}", TestOptions.Regular.WithLanguageVersion(LanguageVersion.CSharp9));
+        }
+
+        [WorkItem(38102, "https://github.com/dotnet/roslyn/issues/38102")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseThrowExpression)]
+        public async Task TestKeepCommnents3()
+        {
+            await TestAsync(
+@"using System;
+
+class Program
+{
+    static void Main() { }
+
+    private object _arg;
+
+    public Program(object arg)
+    {
+        if (arg == null)
+        { 
+            [|throw|] new ArgumentNullException(nameof(arg));// Oh no
```

Comment:
we should add a test to make sure we do something reasonable if we have a comment in both places (on if statement and on the throw)